ValheimPlus/PlayerStats.cs

```diff
@@ -29,89 +29,111 @@ private static void Postfix(Player __instance)
-                        v = v - (v * (Configuration.Current.StaminaUsage.bows) / 100);
-                        break;
-                    case "Unarmed":
-                        v = v - (v * (Configuration.Current.StaminaUsage.unarmed) / 100);
-                        break;
-                    case "Pickaxes":
-                        v = v - (v * (Configuration.Current.StaminaUsage.pickaxes) / 100);
-                        break;
-                    default:
-                        break;
+                    if (weaponType != "")
+                    {
+                        switch (weaponType)
+                        {
+                            case "Swords":
+                                __result -= __result * Configuration.Current.StaminaUsage.swords / 100;
+                                break;
+                            case "Knives":
+                                __result -= __result * Configuration.Current.StaminaUsage.knives / 100;
+                                break;
+                            case "Clubs":
+                                __result -= __result * Configuration.Current.StaminaUsage.clubs / 100;
+                                break;
+                            case "Polearms":
+                                __result -= __result * Configuration.Current.StaminaUsage.polearms / 100;
+                                break;
+                            case "Spears":
+                                __result -= __result * Configuration.Current.StaminaUsage.spears / 100;
+                                break;
+                            case "Axes":
+                                __result -= __result * Configuration.Current.StaminaUsage.axes / 100;
+                                break;
+                            case "Bows":
+                                __result -= __result * Configuration.Current.StaminaUsage.bows / 100;
+                                break;
+                            case "Unarmed":
+                                __result -= __result * Configuration.Current.StaminaUsage.unarmed / 100;
+                                break;
+                            case "Pickaxes":
+                                __result -= __result * Configuration.Current.StaminaUsage.pickaxes / 100;
+                                break;
+                            default:
+                                break;
+                        }
+                    }       
                 }
+            }
+        }
+    }
 
-                if (isHammer)
+    [HarmonyPatch(typeof(Player), "UseStamina")]
+    public static class ChangeStaminaUsageOfTools
+    {
+        private static void Prefix(ref Player __instance, ref float v)
+        {
+            if (Configuration.Current.StaminaUsage.IsEnabled)
+            {
+                string methodName = new StackTrace().GetFrame(2).GetMethod().Name;
+                if (methodName == "UpdatePlacement" || methodName == "Repair")
```

Comment:
instead of these strings, could we run something like `methodName.Equals(nameof(Player.UpdatePlacement))`?

Reply:
Reviewing this section led me to discover an oversight with blocking which I fixed. Previously only affected blocking with a shield and not with any other weapon.
Thank you 😄